forge-game/src/main/java/forge/game/CardTraitBase.java

```diff
@@ -235,6 +236,13 @@ public boolean matchesValid(final Object o, final String[] valids, final Card sr
                     return true;
                 }
             }
+        } else if (o instanceof GameLossReason) {
+            for (String s : valids) {
+                GameLossReason valid = GameLossReason.smartValueOf(s);
+                if (((GameLossReason) o).name().equals(valid.name())) {
+                    return true;
+                }
+            }
```

Comment:
@tool4ever I really need to find a better idea for this, to match an Enum against some valid strings

One main problem is, that can't use Abstract code inside enum classes, so i might need to move that to a helper?

Reply:
`((GameLossReason) o).name().equals(s)` should also work already?

Reply:
yeah, maybe equals IgnoreCase

my idea was more to find a way to not have all EnumTypes I want to check there

like some kind of:
```
enum GameLossReason implements IValidEnum
{
   public bool isValid(String valids[])
}
```

then in the MatchesValid, i could do:
```
} else if (o instanceof IValidEnum) {
  return ((IValidEnum)o).isValid(valids);
}
```

Reply:
using Enum as class name should make it work for others?